### PR TITLE
Fix S5122 FN: Support raw string literals interpolation

### LINKED ISSUE #6383: Fix S5122 FN: Support raw string literals interpolation
### Description

FN for S5122 (PermissiveCors).

Using non constant string interpolation in C#11 raw string literals, the rule is not raising an issue.

### Repro steps

```csharp
           const string part1 = "Access-Control";
           const string part2 = "Allow-Origin";
           Response.Headers.Add($$"""{{part1}}-{{part2}}""", "*"); // FN
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.

### PR DESCRIPTION
**Pull request by cristian-ambrosini-sonarsource**
Fixes #6383

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/PermissiveCors.CSharp11.cs [hunk 1]
```diff
@@ -14,7 +14,11 @@ public void Index(string header, string headerValue)
             string part2 = "Allow-Origin";
             // The Access-Control-Allow-Origin response header indicates whether the response can be shared with requesting code from the given origin.
             Response.Headers.Add("""Access-Control-Allow-Origin""", """*"""); // Noncompliant
-            Response.Headers.Add($$"""{{part1}}-{{part2}}""", "*"); // FN
+            Response.Headers.Add($$"""{{part1}}-{{part2}}""", "*"); // FN (at the moment we validate only constant string)
+
+            const string part3 = "Access-Control";
+            const string part4 = "Allow-Origin";
+            Response.Headers.Add($$"""{{part3}}-{{part4}}""", "*"); // Noncompliant
```
Comment (gregory-paidis-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/PermissiveCors.CSharp11.cs:21]: Maybe add some more testcases with raw "components", like the following:
```csharp
        const string part3 = """Access-Control""";
        const string part4 = """Allow-Origin""";
        Response.Headers.Add($$"""{{part3}}-{{part4}}""", "*"); // Noncompliant

        const string RawStar = """*""";
        Response.Headers.Add($$"""Access-Control-Allow-Origin""", RawStar); // Noncompliant
```

**Review by gregory-paidis-sonarsource — CHANGES_REQUESTED** [review]
LGTM, added a minor comment.

**Review by gregory-paidis-sonarsource — APPROVED** [review]

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/PermissiveCors.CSharp11.cs [hunk 2]
```diff
@@ -10,11 +10,23 @@ public class CorsEnabledManualAddedHeadersController : Controller
...
+            string accessControl = "Access-Control";
+            string allowOrigin = "Allow-Origin";
             // The Access-Control-Allow-Origin response header indicates whether the response can be shared with requesting code from the given origin.
             Response.Headers.Add("""Access-Control-Allow-Origin""", """*"""); // Noncompliant
-            Response.Headers.Add($$"""{{part1}}-{{part2}}""", "*"); // FN
+            Response.Headers.Add($$"""{{accessControl}}-{{allowOrigin}}""", "*"); // FN (at the moment we validate only constant string)
+
+            const string constAccessControl = "Access-Control";
+            const string constAllowOrigin = "Allow-Origin";
+            Response.Headers.Add($$"""{{constAccessControl}}-{{constAllowOrigin}}""", "*"); // Noncompliant
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/PermissiveCors.CSharp11.cs:21]: Could you please also add a C#10 test case where you create a constant interpolated string and pass that as an argument? That should also work, just to document the behavior.

**Review by csaba-sagi-sonarsource — CHANGES_REQUESTED** [review]
I suggested to add one more test case.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/PermissiveCors.CSharp10.cs [hunk 3]
```diff
@@ -23,5 +24,18 @@ public void Bar(IEnumerable<int> collection)
...
+        
+        [ApiController]
+        [Route("[controller]")]
+        public class ConstantInterpolatedStringController : Controller
+        {
+            [HttpGet]
+            public void Index()
+            {
+                const string constAccessControl = "Access-Control";
+                const string constAllowOrigin = "Allow-Origin";
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/PermissiveCors.CSharp10.cs:36]: This test case is also good, and I suggest to add something else like:
```
const string constAccessControl = "Access-Control";
const string constAllowOrigin = "Allow-Origin";
const string interpolatedString = $"{constAccessControl}-{constAllowOrigin}";
Response.Headers.Add(interpolatedString , "*");
```

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM, I left one more comment to add another test case.

**Pull request merged**